### ISSUE TITLE
[BUG]: Getting invalid api-version when using Azcli@2

### ISSUE THREAD

**Issue by jadcosta** [microsoft/azure-pipelines-tasks#18754]
### Task name

azcli@2

### Environment type (Please select at least one enviroment where you face this issue)

- [X] Self-Hosted
- [ ] Microsoft Hosted
- [ ] VMSS Pool
- [ ] Container

### Azure DevOps Server type

dev.azure.com (formerly visualstudio.com)

### Operation system

Windows Server 2019 Datacenter

### Task log

```shell
2023-08-01T16:45:26.3134826Z ##[section]Starting: Add Secret Access Policies to Function APP to Shared Key Vault
2023-08-01T16:45:26.4534910Z ==============================================================================
2023-08-01T16:45:26.4535274Z Task         : Azure CLI
2023-08-01T16:45:26.4535466Z Description  : Run Azure CLI commands against an Azure subscription in a PowerShell Core/Shell script when running on Linux agent or PowerShell/PowerShell Core/Batch script when running on Windows agent.
2023-08-01T16:45:26.4536023Z Version      : 2.225.0
2023-08-01T16:45:26.4536293Z Author       : Microsoft Corporation
2023-08-01T16:45:26.4536500Z Help         : https://docs.microsoft.com/azure/devops/pipelines/tasks/deploy/azure-cli
2023-08-01T16:45:26.4536816Z ==============================================================================
2023-08-01T16:45:27.4223206Z [command]C:\Windows\system32\cmd.exe /D /S /C ""C:\Program Files (x86)\Microsoft SDKs\Azure\CLI2\wbin\az.cmd" --version"
2023-08-01T16:45:30.3155123Z azure-cli                         2.51.0
2023-08-01T16:45:30.3172309Z 
2023-08-01T16:45:30.3181018Z core                              2.51.0
2023-08-01T16:45:30.3198907Z telemetry                          1.1.0
2023-08-01T16:45:30.3214372Z 
2023-08-01T16:45:30.3233305Z Dependencies:
2023-08-01T16:45:30.3246044Z msal                            1.24.0b1
2023-08-01T16:45:30.3296830Z azure-mgmt-resource             23.1.0b2
2023-08-01T16:45:30.3329283Z 
2023-08-01T16:45:30.3335442Z Python location 'C:\Program Files (x86)\Microsoft SDKs\Azure\CLI2\python.exe'
2023-08-01T16:45:30.3378028Z Extensions directory 'C:\Users\svc_nexus_ado_agent\.azure\cliextensions'
2023-08-01T16:45:30.3391152Z 
2023-08-01T16:45:30.3411308Z Python (Windows) 3.10.10 (tags/v3.10.10:aad5f6a, Feb  7 2023, 17:05:00) [MSC v.1929 32 bit (Intel)]
2023-08-01T16:45:30.3417277Z 
2023-08-01T16:45:30.3436061Z Legal docs and information: aka.ms/AzureCliLegal
2023-08-01T16:45:30.3443903Z 
2023-08-01T16:45:30.3462615Z 
2023-08-01T16:45:30.3471036Z Your CLI is up-to-date.
2023-08-01T16:45:30.3706231Z Setting AZURE_CONFIG_DIR env variable to: C:\agent01\_work\_temp\.azclitask
2023-08-01T16:45:30.3853397Z Setting active cloud to: AzureCloud
2023-08-01T16:45:30.4147200Z [command]C:\Windows\system32\cmd.exe /D /S /C ""C:\Program Files (x86)\Microsoft SDKs\Azure\CLI2\wbin\az.cmd" cloud set -n AzureCloud"
2023-08-01T16:45:32.9050155Z [command]C:\Windows\system32\cmd.exe /D /S /C ""C:\Program Files (x86)\Microsoft SDKs\Azure\CLI2\wbin\az.cmd" login --service-principal -u *** "--password=***" --tenant dfbcc178-bccf-4595-8f8e-3a3175df90b7 --allow-no-subscriptions"
2023-08-01T16:45:38.0640267Z [
2023-08-01T16:45:38.0640675Z   {
2023-08-01T16:45:38.0641044Z     "cloudName": "AzureCloud",
2023-08-01T16:45:38.0641521Z     "homeTenantId": "dfbcc178-bccf-4595-8f8e-3a3175df90b7",
2023-08-01T16:45:38.0642046Z     "id": "bab9ed05-2c6e-4631-a0cf-7373c33838cc",
2023-08-01T16:45:38.0642473Z     "isDefault": true,
2023-08-01T16:45:38.0642844Z     "managedByTenants": [],
2023-08-01T16:45:38.0643274Z     "name": "Hiscox Platform Subscription",
2023-08-01T16:45:38.0643684Z     "state": "Enabled",
2023-08-01T16:45:38.0644128Z     "tenantId": "dfbcc178-bccf-4595-8f8e-3a3175df90b7",
2023-08-01T16:45:38.0644544Z     "user": {
2023-08-01T16:45:38.0645049Z       "name": "***",
2023-08-01T16:45:38.0645421Z       "type": "servicePrincipal"
2023-08-01T16:45:38.0645760Z     }
2023-08-01T16:45:38.0646047Z   },
2023-08-01T16:45:38.0646328Z   {
2023-08-01T16:45:38.0646650Z     "cloudName": "AzureCloud",
2023-08-01T16:45:38.0647115Z     "homeTenantId": "dfbcc178-bccf-4595-8f8e-3a3175df90b7",
2023-08-01T16:45:38.0647635Z     "id": "474bd936-ce18-4c16-8f11-3a532a4fa1bf",
2023-08-01T16:45:38.0648066Z     "isDefault": false,
2023-08-01T16:45:38.0648437Z     "managedByTenants": [
2023-08-01T16:45:38.0648762Z       {
2023-08-01T16:45:38.0649170Z         "tenantId": "2f4a9838-26b7-47ee-be60-ccc1fdec5953"
2023-08-01T16:45:38.0649573Z       }
2023-08-01T16:45:38.0649864Z     ],
2023-08-01T16:45:38.0653469Z     "name": "Hiscox Europe Dev/Test",
2023-08-01T16:45:38.0653861Z     "state": "Enabled",
2023-08-01T16:45:38.0654311Z     "tenantId": "dfbcc178-bccf-4595-8f8e-3a3175df90b7",
2023-08-01T16:45:38.0654724Z     "user": {
2023-08-01T16:45:38.0655190Z       "name": "***",
2023-08-01T16:45:38.0655563Z       "type": "servicePrincipal"
2023-08-01T16:45:38.0655905Z     }
2023-08-01T16:45:38.0656197Z   },
2023-08-01T16:45:38.0656475Z   {
2023-08-01T16:45:38.0656810Z     "cloudName": "AzureCloud",
2023-08-01T16:45:38.0657277Z     "homeTenantId": "dfbcc178-bccf-4595-8f8e-3a3175df90b7",
2023-08-01T16:45:38.0657792Z     "id": "4c3b88c7-b886-469d-b9a8-fce396365919",
2023-08-01T16:45:38.0658229Z     "isDefault": false,
2023-08-01T16:45:38.0658600Z     "managedByTenants": [],
2023-08-01T16:45:38.0659038Z     "name": "Hiscox Europe Integration Dev/Test",
2023-08-01T16:45:38.0659452Z     "state": "Enabled",
2023-08-01T16:45:38.0659894Z     "tenantId": "dfbcc178-bccf-4595-8f8e-3a3175df90b7",
2023-08-01T16:45:38.0660314Z     "user": {
2023-08-01T16:45:38.0660736Z       "name": "***",
2023-08-01T16:45:38.0661105Z       "type": "servicePrincipal"
2023-08-01T16:45:38.0661444Z     }
2023-08-01T16:45:38.0661730Z   },
2023-08-01T16:45:38.0662009Z   {
2023-08-01T16:45:38.0662346Z     "cloudName": "AzureCloud",
2023-08-01T16:45:38.0662814Z     "homeTenantId": "dfbcc178-bccf-4595-8f8e-3a3175df90b7",
2023-08-01T16:45:38.0663330Z     "id": "0e2fe536-28d1-4b76-8fa7-73618327c212",
2023-08-01T16:45:38.0663754Z     "isDefault": false,
2023-08-01T16:45:38.0664127Z     "managedByTenants": [],
2023-08-01T16:45:38.0664565Z     "name": "Hiscox Integration Platform Dev/Test",
2023-08-01T16:45:38.0665544Z     "state": "Enabled",
2023-08-01T16:45:38.0665993Z     "tenantId": "dfbcc178-bccf-4595-8f8e-3a3175df90b7",
2023-08-01T16:45:38.0666435Z     "user": {
2023-08-01T16:45:38.0666867Z       "name": "***",
2023-08-01T16:45:38.0667239Z       "type": "servicePrincipal"
2023-08-01T16:45:38.0667585Z     }
2023-08-01T16:45:38.0667865Z   }
2023-08-01T16:45:38.0668140Z ]
2023-08-01T16:45:38.0677061Z [command]C:\Windows\system32\cmd.exe /D /S /C ""C:\Program Files (x86)\Microsoft SDKs\Azure\CLI2\wbin\az.cmd" account set --subscription 4c3b88c7-b886-469d-b9a8-fce396365919"
2023-08-01T16:45:39.2917045Z [command]C:\Windows\System32\WindowsPowerShell\v1.0\powershell.exe -NoLogo -NoProfile -NonInteractive -ExecutionPolicy Unrestricted -Command ". 'C:\agent01\_work\_temp\azureclitaskscript1690908327357.ps1'"
2023-08-01T16:45:46.6698391Z ERROR: (InvalidApiVersionParameter) The api-version '2023-02-01' is invalid. The supported versions are '2023-07-01-preview,2023-03-01-preview,2022-12-01,2022-11-01-preview,2022-09-01,2022-06-01,2022-05-01,2022-03-01-preview,2022-01-01,2021-04-01,2021-01-01,2020-10-01,2020-09-01,2020-08-01,2020-07-01,2020-06-01,2020-05-01,2020-01-01,2019-11-01,2019-10-01,2019-09-01,2019-08-01,2019-07-01,2019-06-01,2019-05-10,2019-05-01,2019-03-01,2018-11-01,2018-09-01,2018-08-01,2018-07-01,2018-06-01,2018-05-01,2018-02-01,2018-01-01,2017-12-01,2017-08-01,2017-06-01,2017-05-10,2017-05-01,2017-03-01,2016-09-01,2016-07-01,2016-06-01,2016-02-01,2015-11-01,2015-01-01,2014-04-01-preview,2014-04-01,2014-01-01,2013-03-01,2014-02-26,2014-04'.
2023-08-01T16:45:46.6707151Z Code: InvalidApiVersionParameter
2023-08-01T16:45:46.6871756Z Message: The api-version '2023-02-01' is invalid. The supported versions are '2023-07-01-preview,2023-03-01-preview,2022-12-01,2022-11-01-preview,2022-09-01,2022-06-01,2022-05-01,2022-03-01-preview,2022-01-01,2021-04-01,2021-01-01,2020-10-01,2020-09-01,2020-08-01,2020-07-01,2020-06-01,2020-05-01,2020-01-01,2019-11-01,2019-10-01,2019-09-01,2019-08-01,2019-07-01,2019-06-01,2019-05-10,2019-05-01,2019-03-01,2018-11-01,2018-09-01,2018-08-01,2018-07-01,2018-06-01,2018-05-01,2018-02-01,2018-01-01,2017-12-01,2017-08-01,2017-06-01,2017-05-10,2017-05-01,2017-03-01,2016-09-01,2016-07-01,2016-06-01,2016-02-01,2015-11-01,2015-01-01,2014-04-01-preview,2014-04-01,2014-01-01,2013-03-01,2014-02-26,2014-04'.
2023-08-01T16:45:48.1041987Z ##[error]Script failed with exit code: 1
2023-08-01T16:45:48.1296134Z [command]C:\Windows\system32\cmd.exe /D /S /C ""C:\Program Files (x86)\Microsoft SDKs\Azure\CLI2\wbin\az.cmd" account clear"
2023-08-01T16:45:50.7795063Z ##[section]Finishing: Add Secret Access Policies to Function APP to Shared Key Vault
```

### Relevant log output

```shell
ERROR: (InvalidApiVersionParameter) The api-version '2023-02-01' is invalid. The supported versions are '2023-07-01-preview,2023-03-01-preview,2022-12-01,2022-11-01-preview,2022-09-01,2022-06-01,2022-05-01,2022-03-01-preview,2022-01-01,2021-04-01,2021-01-01,2020-10-01,2020-09-01,2020-08-01,2020-07-01,2020-06-01,2020-05-01,2020-01-01,2019-11-01,2019-10-01,2019-09-01,2019-08-01,2019-07-01,2019-06-01,2019-05-10,2019-05-01,2019-03-01,2018-11-01,2018-09-01,2018-08-01,2018-07-01,2018-06-01,2018-05-01,2018-02-01,2018-01-01,2017-12-01,2017-08-01,2017-06-01,2017-05-10,2017-05-01,2017-03-01,2016-09-01,2016-07-01,2016-06-01,2016-02-01,2015-11-01,2015-01-01,2014-04-01-preview,2014-04-01,2014-01-01,2013-03-01,2014-02-26,2014-04'.
Code: InvalidApiVersionParameter
Message: The api-version '2023-02-01' is invalid. The supported versions are '2023-07-01-preview,2023-03-01-preview,2022-12-01,2022-11-01-preview,2022-09-01,2022-06-01,2022-05-01,2022-03-01-preview,2022-01-01,2021-04-01,2021-01-01,2020-10-01,2020-09-01,2020-08-01,2020-07-01,2020-06-01,2020-05-01,2020-01-01,2019-11-01,2019-10-01,2019-09-01,2019-08-01,2019-07-01,2019-06-01,2019-05-10,2019-05-01,2019-03-01,2018-11-01,2018-09-01,2018-08-01,2018-07-01,2018-06-01,2018-05-01,2018-02-01,2018-01-01,2017-12-01,2017-08-01,2017-06-01,2017-05-10,2017-05-01,2017-03-01,2016-09-01,2016-07-01,2016-06-01,2016-02-01,2015-11-01,2015-01-01,2014-04-01-preview,2014-04-01,2014-01-01,2013-03-01,2014-02-26,2014-04'.
```

### Aditional info

```shell
I'm getting the attached error when using the following commands with Azcli@2 task:

Command:
az keyvault set-policy --name "$(keyvault.name)" --object-id "$(function.objectId)" --secret-permissions get list
```

**Comment by thijmencap** [comment]
Having the same issue, since the release of 2.51.0, with the following command:
az keyvault network-rule add --name "$(keyvault.name)" --ip-address "$ip/32"

**Comment by AlexTeslenko** [comment]
Hi all! I have the same issue. Is it hear any timeline for the resolution?

**Comment by nilleb** [comment]
for me, this was potentially related to two things:
1. I was not specifying the `--resource-group`
2. I was passing the object ID to the `--spn` argument - which expects instead a service *name*.

hope this helps!

**Comment by omarw** [comment]
> for me, this was potentially related to two things:
> 
> 1. I was not specifying the `--resource-group`
> 2. I was passing the object ID to the `--spn` argument - which expects instead a service _name_.
> 
> hope this helps!

Adding the resource group fixed it for me, thanks!

**Comment by sbonds** [comment]
Upstream issue: https://github.com/Azure/azure-cli/issues/27239

Fixed in Az CLI 2.52.0+
